### ISSUE TITLE
Docker does not work anymore on WSL 2

### ISSUE THREAD

**Issue by ClementNerma** [docker/for-win#13019]
- [X ] I have tried with the latest version of Docker Desktop
  - [X] I have tried disabling enabled experimental features
  - [ ] I have uploaded Diagnostics

### Actual behavior

Since I updated WSL a few days ago (or since the last Docker Desktop background update if one happened in the meantime), Docker is unable to start.

I uninstalled it, re-installed it, and it doesn't start at all - does not even create any WSL 2 VM.

In the process manager I can see the process keeps running in the background with about 22 MB of RAM usage. It goes up and down a little bit. It ends up closing itself.

### Expected behavior

Docker should start normally.

### Information

* Windows version: Windows 11 22H2 (Home), default channel (non-Insiders) build 22621.674
* WSL version: 0.70.0.0
* WSL kernel version: 5.15.68.1
* WSL distribution: Ubuntu 22.04.1 LTS, on WSL 2
* CPU model: Intel Core i7-9700F

### Output of `& "C:\Program Files\Docker\Docker\resources\com.docker.diagnose.exe" check`

[Pastebin](https://pastebin.com/raw/sa3S4MSK)

### Steps to reproduce the behavior

I don't know how to reproduce the bug since it works correctly on other devices.

**Comment by jdavis-software** [comment]
Same thing for me!

**Comment by ClementNerma** [comment]
> Same thing for me!

I see you also answered on https://github.com/docker/for-win/issues/13013, do you have an error message when you start or none? The other issue has one, mine does not.

**Comment by david-engelmann** [comment]
[Experiencing similar issue which I've documented here](https://github.com/docker/for-win/issues/13020)

**Comment by david-engelmann** [comment]
@ClementNerma @jdavis-software [I was able to find a solution and documented in this comment](https://github.com/docker/for-win/issues/13020#issuecomment-1283055393)

**Comment by HollisTech** [comment]
@david-engelmann solution requires destroying your wsl instance. That isn't an acceptable solution, it's disaster recovery.

Note you could likely avoid the DR approach by just cleaning up specific docker folders as noted here: https://github.com/docker/for-win/issues/13020#issuecomment-1283937945 by @stoffpalette

**Comment by david-engelmann** [comment]
@HollisTech As stated in my issue, I attempted to delete all docker specific folders and didn't have success. I would recommend everyone trying those steps before the DR approach but running the `wsl --unregister docker-desktop-data` and `wsl --unregister docker-desktop` commands were the only thing that helped in my situation.

**Comment by david-engelmann** [comment]
@HollisTech I updated the initial DR solution to link out to the less drastic approach with in the same issue. [You can see it here](https://github.com/docker/for-win/issues/13020#issuecomment-1285447765), please let me know if you think there are steps I've missed or steps that are unnecessary.

**Comment by HollisTech** [comment]
Yeah I don't care about the docker instances, so sure that's ok. I still think the whole approach of Docker Desktop inserting its own versions of docker utilities into target wsl instances is just wrong, but that is a related but separate issue,

**Comment by david-engelmann** [comment]
@HollisTech Yeah that aspect is definitely frustrating, specially once you add VS Code servers in the mixes. Thanks for your feedback!

**Comment by claboran** [comment]
Had the same problem this morning after upgrading to v4.13.0.

com.docker.diagnose.exe check
[2022-10-25T18:07:43.540651500Z][com.docker.diagnose.exe][I] set path configuration to OnHost
Starting diagnostics

[PASS] DD0027: is there available disk space on the host?
[PASS] DD0028: is there available VM disk space?
[PASS] DD0002: does the bootloader have virtualization enabled?
[SKIP] DD0018: does the host support virtualization?
[FAIL] DD0001: is the application running? Docker is not running
[PASS] DD0022: is the Virtual Machine Platform Windows Feature enabled?
[PASS] DD0021: is the WSL 2 Windows Feature enabled?
[PASS] DD0024: is WSL installed?
### [FAIL] DD0025: are WSL distros installed? WSL Distribution docker-desktop is missing

The wsl distro docker-desktop was gone.

I found some errors in the install-cli-log-admin

Version: 4.12.0 (85629)
Sha1: 38dadfbe15d75f37fe5285e8466328291ef133b9
Started on: 2022/10/25 05:29:14.377
Resources: C:\Program Files\Docker\Docker\resources
OS: Windows 10 Enterprise
Edition: Enterprise
Id: 2009
Build: 19044
BuildLabName: 19041.1.amd64fre.vb_release.191206-1406
File: C:\ProgramData\DockerDesktop\install-cli-log-admin.txt
CommandLine: "C:\Program Files\Docker\Docker\InstallerCli.exe" -k -p
You can send feedback, including this log file, at https://github.com/docker/for-win/issues
[2022-10-25T05:29:14.524601900Z][Installer         ][Info   ] Killing processes...
[2022-10-25T05:29:14.540603900Z][Installer         ][Info   ] Killing existing com.docker.service process with PID 7140
[2022-10-25T05:29:18.093106700Z][Cmd               ][Info   ] time="2022-10-25T07:29:18.045275000+02:00" level=fatal msg="The specified service does not exist as an installed service."
[2022-10-25T05:29:18.099115000Z][Installer         ][Info   ] Preparing upgrade...
[2022-10-25T05:29:18.153470100Z][HyperV            ][Info   ] Destroy
[2022-10-25T05:29:18.168499900Z][HyperVInfrastructure][Info   ] Try to remove VM DockerDesktopVM
[2022-10-25T05:29:18.635214900Z][HyperVInfrastructure][Info   ] Successfully remove VM DockerDesktopVM
[2022-10-25T05:29:18.640215700Z][LinuxWSL2Engine   ][Info   ] Destroying distro
[2022-10-25T05:29:19.833455300Z][LinuxWSL2Engine   ][Info   ] Ubuntu-20.04: unable to read docker CLI config file: , stderr: cat: /home/cla/.docker/config.json: No such file or directory
[2022-10-25T05:29:19.834457000Z][Installer         ][Info   ] Prepare upgrade completed successfully
[2022-10-25T05:29:19.834457000Z][Installer         ][Info   ] Fixing permissions...
[2022-10-25T05:29:19.835457300Z][Installer         ][Info   ] Deleting directory C:\Program Files\Docker\cli-plugins
[2022-10-25T05:29:19.839454000Z][Installer         ][Info   ] granting ReadAndExecute to everyone for C:\ProgramData\DockerDesktop
[2022-10-25T05:29:19.842454600Z][Installer         ][Error  ] Error changing security on C:\Program Files\Docker\cli-plugins: Invalid path: C:\Program Files\Docker\cli-plugins
[2022-10-25T05:29:19.842454600Z][Installer         ][Info   ] Permissions should be fixed.

apart from

------------------------------------------------------------------------------->8
Version: 4.13.0 (89412)
Sha1: 
Started on: 2022.10.25 05:29:11.562
Resources: C:\Users\LABORACH\AppData\Local\Docker Desktop Installer\resources
OS: Windows 10 Enterprise
Edition: Enterprise
Id: 2009
Build: 19044
BuildLabName: 19041.1.amd64fre.vb_release.191206-1406
File: C:\ProgramData\DockerDesktop\install-log-admin.txt
CommandLine: "C:\Users\LABORACH\AppData\Local\Docker Desktop Installer\update--2043227176.exe" "install" --fallback-url="https://desktop.docker.com/win/main/amd64/89412/Docker%20Desktop%20Installer.exe" --confirm --auto-close-and-start-app --relaunch-as-admin
You can send feedback, including this log file, at https://github.com/docker/for-win/issues
[2022-10-25T05:29:11.760517400Z][ManifestAndExistingInstallationLoader][Info   ] No install path specified, looking for default installation registry key
[2022-10-25T05:29:12.230563200Z][Installer         ][Info   ] Existing installation found: build=85629, version=4.12.0, basePath=C:\Program Files\Docker\Docker, channelUrl=https://desktop.docker.com/win/main/amd64/appcast.xml
[2022-10-25T05:29:12.501157900Z][InstallWorkflow   ][Info   ] Cancel pending background download
[2022-10-25T05:29:12.502157500Z][BackgroundTransfer][Info   ] Cancel current background transfer job
[2022-10-25T05:29:12.510157800Z][InstallWorkflow   ][Info   ] Using package: res:DockerDesktop
[2022-10-25T05:29:12.516158200Z][InstallWorkflow   ][Info   ] Downloading
[2022-10-25T05:29:13.116349700Z][InstallWorkflow   ][Info   ] Extracting manifest
[2022-10-25T05:29:13.613327400Z][InstallWorkflow   ][Info   ] Manifest found: version=89412, displayVersion=4.13.0, channelUrl=https://desktop.docker.com/win/main/amd64/appcast.xml
[2022-10-25T05:29:13.613327400Z][InstallWorkflow   ][Info   ] Checking prerequisites
[2022-10-25T05:29:13.613327400Z][InstallWorkflow   ][Info   ] Patch installer
[2022-10-25T05:29:14.016980900Z][InstallWorkflow   ][Info   ] Uninstalling previous version in update mode
[2022-10-25T05:29:14.023980200Z][UninstallWorkflow ][Info   ] Cancel pending background download
[2022-10-25T05:29:14.023980200Z][BackgroundTransfer][Info   ] Cancel current background transfer job
[2022-10-25T05:29:14.028980900Z][UninstallWorkflow ][Info   ] Running before update actions
[2022-10-25T05:29:14.030007300Z][UninstallWorkflow ][Info   ] Proceeding with CommunityInstaller.ExecAction
[2022-10-25T05:29:14.057982400Z][Installer         ][Warning] Failed to track the installer started event
[2022-10-25T05:29:19.873683900Z][UninstallWorkflow ][Info   ] Proceeding with CommunityInstaller.KillLockingProcessesAction
[2022-10-25T05:29:20.516714200Z][UninstallWorkflow ][Info   ] Proceeding with installed components
[2022-10-25T05:29:20.517715500Z][UninstallWorkflow ][Info   ] Proceeding with CommunityInstaller.ExecAction
[2022-10-25T05:29:20.517715500Z][UninstallWorkflow ][Info   ] Proceeding with CommunityInstaller.PathAction
[2022-10-25T05:29:20.837470400Z][UninstallWorkflow ][Info   ] Proceeding with CommunityInstaller.PathAction
[2022-10-25T05:29:21.153989900Z][UninstallWorkflow ][Info   ] Proceeding with CommunityInstaller.AutoStartAction
[2022-10-25T05:29:21.154990000Z][UninstallWorkflow ][Info   ] Proceeding with CommunityInstaller.ShortcutAction
[2022-10-25T05:29:21.155999600Z][UninstallWorkflow ][Info   ] Proceeding with CommunityInstaller.ShortcutAction
[2022-10-25T05:29:21.157040000Z][UninstallWorkflow ][Info   ] Proceeding with CommunityInstaller.ServiceAction
[2022-10-25T05:29:21.161991800Z][UninstallWorkflow ][Info   ] Proceeding with CommunityInstaller.EnableFeaturesAction
[2022-10-25T05:29:21.161991800Z][UninstallWorkflow ][Info   ] Proceeding with CommunityInstaller.EnableFeaturesAction
[2022-10-25T05:29:21.161991800Z][UninstallWorkflow ][Info   ] Proceeding with CommunityInstaller.AddToGroupAction
[2022-10-25T05:29:21.163049000Z][UninstallWorkflow ][Info   ] Proceeding with CommunityInstaller.CreateGroupAction
[2022-10-25T05:29:21.163049000Z][UninstallWorkflow ][Info   ] Unregistering product
[2022-10-25T05:29:21.167047700Z][RegisterProductStep][Error  ] Failed to delete registry key: Eine Unterschlüsselstruktur kann nicht gelöscht werden, da der Unterschlüssel nicht vorhanden ist.
[2022-10-25T05:29:21.302027900Z][UninstallWorkflow ][Info   ] Uninstalled finished

Not sure what do now.

Any help is appreciated...

**Comment by david-engelmann** [comment]
@claboran What are the results when you run the `wsl -l -v` command?

**Comment by claboran** [comment]
@david-engelmann
  Ubuntu-20.04           Stopped         2
  docker-desktop-data    Stopped         2

**Comment by david-engelmann** [comment]
@claboran If docker-desktop said stopped/uninstalling run the `wsl --unregister docker-desktop`, if that doesn't work also try the `wsl --unregister  docker-desktop-data`. If Ubuntu isn't running try restarting it and making sure it's the distro referenced by docker

**Comment by claboran** [comment]
@david-engelmann Thanks for the help will give a try today.

**Comment by HollisTech** [comment]
An example of this problem is entirely reproducible.
Consider a wsl instance with a docker compose based service that autostarts.


Scenario 1: (most common) DDW starts before wsl instance. It tries but fails to autostart the docker service, as the instance is un available.

Scenario 2: wsl instance starts before docker desktop for windows.  Once DDW starts everything works correctly.

The point is that DDW can have dependencies on specific WSL instances, but ignores those dependencies on startup. Perhaps it ought to have an option to start the wsl instance(s) it depends on?

 /lifecycle frozen
